### ISSUE TITLE
DialogFragment and onCreateDialog

### ISSUE THREAD

**Issue by nicomott** [Prototik/HoloEverywhere#120]
It seems that we can't display a DialogFragment that uses onCreateDialog to display an AlertDialog. I added a button in the demo to display a DialogFragment.

```
public class TestDialogFragment extends SDialogFragment {
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Builder b = new Builder(getActivity());
        b.setCancelable(true);
        b.setMessage("Test message")
        .setTitle("Test Title")
        .setNeutralButton(android.R.string.ok, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialog, int which) {
                dismiss();
            }
        });
        return b.create();
    }
}
```

In Demo Activitity : 

```
public void showAlertDialogFragment(View v) {
    TestDialogFragment newFragment = new TestDialogFragment();
    newFragment.show();
}
```

It gets an NPE on getSupportFragmentManager() in Fragment class. If a give a FragmentTransaction object in the show() method, it doesn't crash, but nothing is displayed.

**Comment by Prototik** [comment]
Try... Try to call replace() with fragment manager...
I... I don't know. I haven't pc in this moment.

**Comment by nicomott** [comment]
Yes, I should have said it, I tried every methods on DialogFragment, nothing works.

**Comment by nicomott** [comment]
Ok so I think that the bug in is DialogFragment class, line 181. We try to get the inherited field "mContainerId" like this: 

```
Field field = getClass().getField("mContainerId");
```

It will always throw a NoSuchFieldException, because it's not declared in this class, but in the superclass android.support.v4.app.Fragment.

**Comment by tverona** [comment]
It seems a little scary to rely on a private field in a framework class, no?

**Comment by nicomott** [comment]
I don't think there is a better way, but I may be wrong. 
Anyway, I fixed that with this code, maybe someone will have something cleaner later : 

```
Class<?> supportFragmentClass = getClass().getSuperclass().getSuperclass().getSuperclass().getSuperclass();
Field field = supportFragmentClass.getDeclaredField("mContainerId");
```

**Comment by Prototik** [comment]
Fixed in 1.3, without reflection :)
